File: VB/WindowsFormsApplication3/CustomTreeListHandler.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraTreeList
Imports DevExpress.XtraTreeList.Handler
Imports DevExpress.XtraTreeList.Nodes
Imports DevExpress.XtraTreeList.ViewInfo
Imports DevExpress.Utils.Drawing
Imports DevExpress.XtraTreeList.Painter
Imports System.Reflection

Namespace WindowsFormsApplication3
	Public Class CustomTreeListHandler
		Inherits TreeListHandler
		Public Sub New(ByVal treeList As TreeList)
			MyBase.New(treeList)
		End Sub

		Protected Overrides Function CreateState(ByVal state As TreeListState) As TreeListHandler.TreeListControlState
            If state = TreeListState.NodePressed Then
                Return New CustomPressedState(Me)
            End If
            If state = TreeListState.NodeDragging Then
                Return New CustomNodeDraggingState(Me)
            End If
			Return MyBase.CreateState(state)
		End Function


	End Class
	Public Class CustomNodeDraggingState
		Inherits DevExpress.XtraTreeList.Handler.TreeListHandler.NodeDraggingState
		Public Sub New(ByVal handler As TreeListHandler)
			MyBase.New(handler)
		End Sub
		Protected Overrides Sub DoStartNodeDragging(ByVal ri As RowInfo, ByVal drgevent As DragEventArgs)
			'base.DoStartNodeDragging(rin, drgevent);
		End Sub
	End Class
End Namespace

File: CS/WindowsFormsApplication3/CustomTreeListHandler.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraTreeList;
using DevExpress.XtraTreeList.Handler;
using DevExpress.XtraTreeList.Nodes;
using DevExpress.XtraTreeList.ViewInfo;
using DevExpress.Utils.Drawing;
using DevExpress.XtraTreeList.Painter;
using System.Reflection;

namespace WindowsFormsApplication3 {
    public class CustomTreeListHandler : TreeListHandler {
        public CustomTreeListHandler(TreeList treeList) : base(treeList) { }

        protected override TreeListHandler.TreeListControlState CreateState(TreeListState state) {
            if (state == TreeListState.NodePressed) {
                return new CustomPressedState(this);
            }
            if (state == TreeListState.NodeDragging) {
                return new CustomNodeDraggingState(this);
            }
            return base.CreateState(state);
        }


    }
    public class CustomNodeDraggingState : DevExpress.XtraTreeList.Handler.TreeListHandler.NodeDraggingState {
        public CustomNodeDraggingState(TreeListHandler handler) : base(handler) { }
        protected override void DoStartNodeDragging(RowInfo ri, DragEventArgs drgevent) {
            //base.DoStartNodeDragging(rin, drgevent);
        }
    }
}
